### ISSUE TITLE
How to use LBFGS in Pytorch-Lightening 

### ISSUE THREAD

**Issue by peymanpoozesh** [PyTorchLightning/pytorch-lightning#3672]
how to use the LBFGS in lightening. the loss in the  following code does not change and it seems that LBFGS is not be
being used correctly 

```
def configure_optimizers(self):
   optimizer = optim.LBFGS(self.parameters(), lr=self.hparams.lr_LBFGS)
  return optimizer


def training_step(self,train_batch,batch_idx):
     x,t=train_batch
     lg, lb, li = self.problem_formulation(x, t,self.x_Array,self.t_Array,self.bndry,self.pi)
     loss=lg+lb+li
     self.lg=lg
     self.lb=lb
     self.li=li
     return {'loss':loss,'lg':lg, 'lb':lb, 'li':li}


def backward(self, trainer, loss, optimizer, optimizer_idx):
     loss.backward(retain_graph=True)

def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, second_order_closure, 
	on_tpu=True, using_native_amp=False, using_lbfgs=True):
     optimizer.step(second_order_closure)`
```

**Comment by peymankhodro** [comment]
I am trying to use LFBGS optimizer in pytorch-lighteninng. how to use it in pytorch-lightening?

**Comment by rohitgr7** [comment]
let's see what the problem is here https://github.com/PyTorchLightning/pytorch-lightning/issues/4083
